tests/components/sentry/test_init.py

```diff
@@ -0,0 +1,330 @@
+    current_platform_mock.get().domain = "light"
+
+    with patch(
+        "homeassistant.components.sentry.entity_platform.current_platform",
+        new=current_platform_mock,
+    ):
+        result = process_before_send(
+            hass,
+            options={},
+            channel="test",
+            huuid="12345",
+            system_info={"installation_type": "pytest"},
+            custom_components=["ironing_robot"],
+            event={},
+            hint={},
+        )
+
+    assert result
+    assert result["tags"]["integration"] == "hue"
+    assert result["tags"]["platform"] == "light"
+    assert result["tags"]["custom_component"] == "no"
+
+    current_platform_mock.get().platform_name = "ironing_robot"
+    current_platform_mock.get().domain = "switch"
+
+    with patch(
+        "homeassistant.components.sentry.entity_platform.current_platform",
+        new=current_platform_mock,
+    ):
+        result = process_before_send(
+            hass,
+            options={CONF_EVENT_CUSTOM_COMPONENTS: True},
+            channel="test",
+            huuid="12345",
+            system_info={"installation_type": "pytest"},
+            custom_components=["ironing_robot"],
+            event={},
+            hint={},
+        )
+
+    assert result
+    assert result["tags"]["integration"] == "ironing_robot"
+    assert result["tags"]["platform"] == "switch"
+    assert result["tags"]["custom_component"] == "yes"
+
+
+@pytest.mark.parametrize(
+    "logger,tags",
+    [
+        ("adguard", {"package": "adguard"}),
+        (
+            "homeassistant.components.hue.coordinator",
+            {"integration": "hue", "custom_component": "no"},
+        ),
+        (
+            "homeassistant.components.hue.light",
+            {"integration": "hue", "platform": "light", "custom_component": "no"},
+        ),
+        (
+            "homeassistant.components.ironing_robot.switch",
+            {
+                "integration": "ironing_robot",
+                "platform": "switch",
+                "custom_component": "yes",
+            },
+        ),
+        (
+            "homeassistant.components.ironing_robot",
+            {"integration": "ironing_robot", "custom_component": "yes"},
+        ),
+        ("homeassistant.helpers.network", {"helpers": "network"}),
+        ("tuyapi.test", {"package": "tuyapi"}),
+    ],
+)
+async def test_logger_event_extraction(hass: HomeAssistant, logger, tags):
+    """Test extraction of information from Sentry logger events."""
+
+    result = process_before_send(
+        hass,
+        options={
+            CONF_EVENT_CUSTOM_COMPONENTS: True,
+            CONF_EVENT_THIRD_PARTY_PACKAGES: True,
+        },
+        channel="test",
+        huuid="12345",
+        system_info={"installation_type": "pytest"},
+        custom_components=["ironing_robot"],
+        event={"logger": logger},
+        hint={},
+    )
+
+    assert result
+    assert result["tags"] == {
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6042836)